UnityProject/Assets/Scripts/Core/Input System/InteractionV2/TileInteraction/PipeDeconstruction.cs

```diff
@@ -48,7 +48,7 @@ public override void ServerPerformInteraction(TileApply interaction)
 
 			for (var i = 0; i < metaDataNode.PipeData.Count; i++)
 			{
-				if (metaDataNode.PipeData[i].RelatedTile != pipeTile) continue;
+				if (metaDataNode.PipeData[i].RelatedTile != pipeTile) continue; //TODO Stuff like layers and stuff can be included
```

Comment:
what does the TODO mean?

Reply:
oh yeah, because pipe layers checking the tile is a same will return incorrect results if they're different layered pipes since they will be still the same tile , just using tile offsets so, it's best to use exact location that will be planned in future as vector4int (xyz for 3D  and then w for the stuff in between layers like underfloor )